src/Database/MorphPivot.php

```diff
@@ -0,0 +1,190 @@
+<?php namespace Winter\Storm\Database;
+
+/**
+ * Reimplementation of \Illuminate\Database\Eloquent\Relations\MorphPivot
+ * but extends Winter's Pivot base class instead of Laravel's to get access to
+ * Winter magic.
+ */
+class MorphPivot extends Pivot
```

Comment:
@bennothommo I've added a [comment](https://github.com/laravel/framework/pull/25851#issuecomment-1133820077) to the original AsPivot PR making a case for the addition of an AsMorphPivot trait, we could potentially submit the PR ourselves to simplify our codebase.